### DISCUSSION TITLE
20240321 Weekly Development Meeting

### DISCUSSION THREAD

**Discussion by josecelano** [nautilus-cyberneering/.github#87] [Meetings]
- Time:  9:30 in Canary Islands (WET (UTC+0)).
- Time: 10:30 in Vaduz (CET (UTC+1)).

Location:
<https://meeting.blockfinance-eco.li/NautilusCyberneeringWeeklyDevMeeting>

Expected duration: 2 to 4 hours with breaks

Communicated attendants:

- CGarnham(@da2ce7)  - 
- JCelano(@josecelano) - x
- CBosse(@cgbosse) - x
- MMelchor(@mario-nt) - x
- GByrne(@grmbyrn) - x
- MVanDijke(@mickvandijke) - 
- Iván Santana(@ivan9989-pkj) - x
- Ismael Torrado(@Lepisma07) - x

-------------------------------------------------------------

## Agenda:

1. Everyone will take their turn and:
- Go over his notes added as comments for the meeting which will have been added in advance to give an update of all the work done during the previous week indicating.
	- The issues worked on
	- The progress made on them
	- Other things done
- Inform of any faced challenges and request help if necessary to work on specific issues.
- Highlight the encountered difficulties, especially if there is a risk of them causing delays or affecting end product's quality.
- Communicate what will be the next week's tasks.
- List any new topics for further discussion in an additional Technical meeting that he noted for future discussion in comments to this discussion either before the meeting or after his turn so that they get considered.

During the meeting every team member will take their turns and follow this sequence.
The topics for further discussion will be noted in advance by each the team member and to the meetings agenda.

2. At the end of the meeting if there any new technical topics to be discussed further it will be decided to
set another meeting date for the discussion of these or to hold the meeting hereafter.

Also the persons to be part of the meeting will be decided on so that the others will be free to continue their own work.

**Comment by josecelano** [comment]
### Torrust Tracker

#### Benchmarking

There was an open issue "[Create benchmarking tool/test for UDP tracker](https://github.com/torrust/torrust-tracker/issues/87)".

- [x] https://github.com/torrust/torrust-tracker/issues/87

I decided to try Aquatic benchmarking tools.

1. Reviewed the current torrent repository benchmarking.
2. Tried the Aquatic UDP load test tool.
3. Added Torrust Tracker to the Aquatic bencher: https://github.com/greatest-ape/aquatic/pull/191. Pending to make some changes like providing the minimum configuration as possible.

It's all summarized in a new blog post: [Benchmarking the Torrust BitTorrent Tracker](https://torrust.com/benchmarking-the-torrust-bittorrent-tracker).

[Benchmarking docs](https://github.com/torrust/torrust-tracker/blob/develop/docs/benchmarking.md) were also updated.

#### Profiling

I've been learning how to do profiling in Rust mainly watching videos on YouTube. I will add the links to them in a new blog post.

My main conclusion is that it's hard to do profiling for async code. You only see a lot of metrics for Tokio functions.

I've been using two approaches:

- [valgrind](https://valgrind.org/) and [kcachegrind](https://kcachegrind.github.io/).
- [flamegraph](https://crates.io/crates/flamegraph). It relies on [perf](https://perf.wiki.kernel.org/index.php/Main_Page) on Linux.

Issues: 

- [ ] https://github.com/torrust/torrust-tracker/issues/746

#### Next

- Continue profiling with "flamegraph" or other custom profiling tools.
- [Write also a new blog post](https://github.com/torrust/torrust-website/issues/154).

**Comment by grmbyrn** [comment]
## Learning
* Vue - [Vue Expenses Tracker](https://github.com/grmbyrn/vue-expenses-tracker)
* Nuxt - [Nuxt Dojo](https://github.com/grmbyrn/nuxt-dojo)

## Frontend Mentor
* [Bookmark Landing Page](https://github.com/grmbyrn/bookmark-landing-page)
